assets/aws/files/install-hardened.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+if [[ "${DEBUG:-false}" == "true" ]]; then
+    set -x
+fi
+
+# Set some curl options so that temporary failures get retried
+# More info: https://ec.haxx.se/usingcurl-timeouts.html
+CURL_OPTS="-L --retry 100 --retry-delay 0 --connect-timeout 10 --max-time 300"
+
+# Update packages
+dnf -y update
+
+# Install 
+#  - uuid used for random token generation,
+#  - python for certbot
+dnf install -y uuid python3
+
+# Install certbot 
+python3 -m venv /opt/certbot
+/opt/certbot/bin/pip install --upgrade pip
+/opt/certbot/bin/pip install certbot certbot-dns-route53
+ln -s /opt/certbot/bin/certbot /usr/bin/certbot
+
+# Create teleport user. It is helpful to share the same UID
+# to have the same permissions on shared NFS volumes across auth servers and for consistency.
+useradd -r teleport -u ${TELEPORT_UID} -d /var/lib/teleport
+# Add teleport to adm group to read and write logs
+usermod -a -G adm teleport
+
+# Setup teleport run dir for pid files
+mkdir -p /run/teleport/ /var/lib/teleport /etc/teleport.d
+chmod 0700 /var/lib/teleport
+chown -R teleport:adm /run/teleport /var/lib/teleport /etc/teleport.d/
+
+# Download and install teleport binaries
+pushd /tmp || exit
+# Install the FIPS version of Teleport if /tmp/teleport-fips is present
+if [ -f /tmp/teleport-fips ]; then
+    TARBALL_FILENAME="/tmp/files/teleport-ent-v${TELEPORT_VERSION}-linux-amd64-fips-bin.tar.gz"
```

Comment:
There's a lot of repetition in this block, where the only real difference is the tarball filename, which can be factored out before, and the tarball directory, which can be ignored with `--strip-components=1` to the tar command. Something like (this assumes $TELEPORT_FIPS is passed in via the hcl config as mentioned elsewhere in this review):
```
# Download and install teleport binaries
case "${TELEPORT_TYPE}-${TELEPORT_FIPS}" in
        oss-0) TARBALL="teleport-v${TELEPORT_VERSION}-linux-amd64-bin.tar.gz" ;;
        ent-0) TARBALL="teleport-ent-v${TELEPORT_VERSION}-linux-amd64-bin.tar.gz" ;;
        ent-1) TARBALL="teleport-ent-v${TELEPORT_VERSION}-linux-amd64-fips-bin.tar.gz" ;;
        oss-1)
                echo "OSS FIPS not supported" >&2
                exit 1
                ;;
esac
TARBALL_FILENAME="/tmp/files/${TARBALL}"

if [[ -f "${TARBALL_FILENAME}" ]]; then
        echo "Found locally uploaded tarball: ${TARBALL_FILENAME}, moving to /tmp/teleport.tar.gz"
        mv "${TARBALL_FILENAME}" /tmp/teleport.tar.gz
else
        echo "Downloading teleport tarball ${TARBALL}"
        curl ${CURL_OPTS} -o /tmp/teleport.tar.gz "https://get.gravitational.com/teleport/${TELEPORT_VERSION}/${TARBALL}"
fi

# Extract tarball to /tmp/teleport to get the binaries out
mkdir /tmp/teleport
tar -C /tmp/teleport -x -z -f /tmp/teleport.tar.gz --strip-components=1
cp /tmp/teleport/{tctl,tsh,teleport} /usr/local/bin
rm -rf /tmp/teleport /tmp/teleport.tar.gz

if [[ "${TELEPORT_FIPS}" == 1 ]]; then
        # add --fips to 'teleport start' commands in FIPS mode
        sed -i -E "s_ExecStart=/usr/local/bin/teleport start(.*)_ExecStart=/usr/local/bin/teleport start --fips\1_g" /etc/systemd/system/teleport*.service
fi

```